openawsem/scripts/mm_run.py

```diff
@@ -79,19 +79,78 @@ def run(args):
     checkpoint_reporter_frequency = 10000
 
 
-
-    snapShotCount = 400
-    stepsPerT = int(args.steps/snapShotCount)
-    Tstart = args.tempStart
-    Tend = args.tempEnd
-    if args.reportFrequency == -1:
-        if stepsPerT == 0:
-            reporter_frequency = 4000
-        else:
-            reporter_frequency = stepsPerT
+    ## Configure number of timesteps and number of frames/recording interval
+    # do some basic checks on user input
+    if args.steps < 1:
+        raise ValueError("Number of timesteps --steps must be a positive integer")
     else:
+        total_steps = args.steps
+    if args.numFrames > total_steps:
+        # the user has asked for more frames than timesteps
+        raise ValueError("Number of frames --numFrames cannot be greater than number of timesteps --steps")
+    elif args.numFrames == 0:
+        # the user has asked us to run a simulation with 0 frames
+        raise ValueError("Number of frames --numFrames cannot be zero")
+    if int(total_steps/args.reportFrequency) == 0: # int trucates so it works like the floor function for positive numbers
+        # the user has asked us to run a simulation with 0 frames
+        raise ValueError("Number of frames implied by --reportFrequency (calculated as NUM_TIMESTEPS/REPORT_FREQUENCY) must be at least 1")
+    if args.reportFrequency != -1 and args.numFrames != -1:
+        # The user has specified both --reportFrequency and --numFrames. We raise an error unless these directions are perfectly unambiguous.
+        if not (total_steps/args.reportFrequency).is_integer():
+            raise ValueError("Number of timesteps --steps is not divisible by the number of frames implied by --reportFrequency (calculated as NUM_TIMESTEPS/REPORT_FREQUENCY).")
+        elif args.numFrames != total_steps/args.reportFrequency:
+            raise ValueError("Number of frames from --numFrames and number of frames implied by --reportFrequency (calculated as NUM_TIMESTEPS/REPORT_FREQUENCY) disagree.\n\
```

Comment:
If user gave both, then set the number of frames to the maximum of both arguments and print a warning.
Alternatively, you can make report Frequency the default value if both are given.